teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -14,113 +14,138 @@ authors:
+all necessary functionality into a single Task. This allows multiple "actions" to be run in one pod, but hurts reusability and makes parallel execution more difficult.
+
+### Use PipelineResources (deprecated) to express a workflow in one Task
+PipelineResources allowed multiple pieces of functionality to run in a single pod by building some of these functions into the TaskRun controller.
+This allowed some workflows to be written as single Tasks.
+For example, the "git" and "image" PipelineResources made it possible to create a workflow that cloned and built a repo, and pushed the resulting image to
+an image repository, all in one Task. 
+However, PipelineResources hurt reusability because they required Task authors to anticipate what other functionality would be needed before and after the Task.
+For this reason, among others, they were deprecated; please see [TEP-0074](./0074-deprecate-pipelineresources.md) for more information.
+
+### Rely on the Affinity Assistant for improved TaskRun scheduling
+The Affinity Assistant schedules TaskRuns that share a PVC on the same node.
+This feature allows TaskRuns that share PVCs to run in parallel in a system that supports only `ReadWriteOnce` Persistent Volumes.
+However, this does not address the underlying issues of pod overhead and the need to shuttle data between TaskRuns in different pods.
+It also comes with its own set of drawbacks, which are described in more detail in
+[TEP-0046: Colocation of Tasks and Workspaces](https://github.com/tektoncd/community/pull/318/files).
+
+### Use Task results to share data without using PVCs
+Tasks may emit string results that can be used as [parameters of subsequent Tasks](https://tekton.dev/docs/pipelines/pipelines/#passing-one-task-s-results-into-the-parameters-or-when-expressions-of-another).
+There is an existing [TEP](https://github.com/tektoncd/community/pull/477/files) for supporting dictionary and array results as well.
+However, results are not designed to handle large, arbitrary forms of data like source repositories or images.
+
+## Goals
 
 - Make it possible to combine Tasks together so that you can run multiple
-  Tasks together and have control over the scheduling overhead (i.e. pods and volumes required) at authoring time
-  in a way that can be reused (e.g. in a Pipeline)
-- Add some of [the features we don't have without PipelineResources](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#)
-  to Tekton Pipelines (without requiring use of PipelineResources), specifically the first feature listed in
-  [the doc](https://docs.google.com/document/d/1KpVyWi-etX00J3hIz_9HlbaNNEyuzP6S986Wjhl3ZnA/edit#heading=h.gi1d1dikb39u):
-  **Task adapters/specialization**
+  Tasks together and have control over the pods and volumes required.
+- Provide a mechanism to colocate Tasks that "run things" (e.g. a build)
```

Comment:
this might be just a vague but instead of "run things" you could say something like "execute a Task's core logic" - still very hand wavy XD but im trying to get at the fact that in these use cases there's usually somethign you're trying to do (e.g. run tests, build an artifact), and you need inputs and outputs to do it, but they arent the core of what you're doing (and to be reusable you need their logic to vary)

Reply:
agreed that this is very hand wavy 😬 but I like the suggested improvement